server.js

```diff
@@ -1,33 +1,107 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+import express from 'express';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+
+import booksData from './data/books.json';
+
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
 
-// Start defining your routes here
+// Routes start here:
+// Home: path /
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.send("Hello world, welcome to Vanessa's Bookish API! - See this API live at: https://vane-bookish-app.netlify.app/")
+});
+
+// Books: path /books - implemented pagination, each page will show 50 books
+// at a time (we have 500 books, so we have a max of 10 pages)
+// path example: /books?p=1
+app.get('/books', (req, res) => {
+  const postCount = booksData.length;
+  const perPage = 50;
+  // pageCount: counts amount of pages we end up with depending on our data's length
+  const pageCount = Math.ceil(postCount / perPage);
+
+  // page: determines value for the p variable in query - path: /books?p=3
+  let page = parseInt(req.query.p);
+  if(page < 1) page = 1;
+  // page value can't go over pageCount (pageCount is max amount of pages we have)
+  if(page > pageCount) page = pageCount;
+
+  // Calcuates which items to show on the page (example from 0 to 50)
+  const from = perPage * (page - 1);
+  let to = page * perPage;
+  if(to < 0) to = 0;
+
+  res.json({
+    books: booksData.slice(from, to),
+    page,
+    pageCount
+  });
+});
+
+// Show books by a specific author - example path: /books/search?author=Dan&20Brown
+// Added toLowerCase to make sure an author is found even if user types the name in lower case
+app.get('/books/search', (req, res) => {
+  const author = req.query.author;
+  const authorBooks = booksData.filter((item) => item.authors.toLowerCase().includes(author.toLowerCase()));
+
+  // If no books are found, the response is an empty array. Use that data to show an error message instead
+  if (authorBooks.length === 0) {
+    res.status(404).json("Sorry, could not find books by that author name :(")
+  };
+  
+  res.json(authorBooks);
+});
```

Comment:
Would be nice if this was included as a query under the `books` endpoint to make it more RESTful!